packages/node-runtime-worker-thread/src/serializer.ts

```diff
@@ -0,0 +1,145 @@
+import v8 from 'v8';
+import { ShellResult } from '@mongosh/shell-evaluator';
+import { inspect } from 'util';
+import { EJSON, Document } from 'bson';
+
+export function serialize(data: unknown): string {
```

Comment:
I think the v8.serialization and the serialization of result types belongs to 2 different layers, i would not put them together.

Honestly i'd put `serialize` and `deserialize` in the rpc.ts directly, it feels those are much more infrastructural and indepentent from the others which are more domain specific.

Reply:
Yeah, good catch, will move it back.

Reply:
Done in d0f3aef